src/Perpetuum/Modules/BlobEmissionModulatorModule.cs

```diff
@@ -61,29 +61,30 @@ protected override void OnAction()
             if (zone == null)
                 return;
 
-            Position targetPosition = new Position();
+            Position? posistion = null;
 
             var myLock = GetLock();
             if (myLock is TerrainLock)
             {
-                targetPosition = (myLock as TerrainLock).Location.AddToZ(BLOB_EMITTER_HEIGHT);
+                posistion = (myLock as TerrainLock).Location.AddToZ(BLOB_EMITTER_HEIGHT);
             }
             else if (myLock is UnitLock)
             {
-                targetPosition = (myLock as UnitLock).Target.CurrentPosition.AddToZ(BLOB_EMITTER_HEIGHT);
+                posistion = (myLock as UnitLock).Target.CurrentPosition.AddToZ(BLOB_EMITTER_HEIGHT);
             }
             else
             {
                 OnError(ErrorCodes.InvalidLockType);
                 return;
             }
 
-            if(targetPosition.Equals(new Position()))
+            if(posistion == null)
```

Comment:
Unless AddToZ can return null, this is impossible.

Reply:
Additionally, in future rather than == null, use `.HasValue` or null coalesce `??` where appropriate